source/UIAHandler/_remoteOps/instructions/arithmetic.py

```diff
@@ -0,0 +1,110 @@
+# A part of NonVisual Desktop Access (NVDA)
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+# Copyright (C) 2023-2024 NV Access Limited
+
+"""
+This module contains the instructions that perform arithmetic operations,
+such as addition, subtraction, multiplication, and division.
+Both binary and in-place operations are supported.
+"""
+from __future__ import annotations
+from dataclasses import dataclass
+from .. import lowLevel
+from .. import builder
+from ._base import _TypedInstruction
+
+
+@dataclass
+class BinaryAdd(_TypedInstruction):
+	opCode = lowLevel.InstructionType.BinaryAdd
+	result: builder.Operand
+	left: builder.Operand
+	right: builder.Operand
+
+	def localExecute(self, registers: dict[lowLevel.OperandId, object]):
+		registers[self.result.operandId] = registers[self.left.operandId] + registers[self.right.operandId]
+
+
+@dataclass
+class BinarySubtract(_TypedInstruction):
+	opCode = lowLevel.InstructionType.BinarySubtract
+	result: builder.Operand
+	left: builder.Operand
+	right: builder.Operand
+
+	def localExecute(self, registers: dict[lowLevel.OperandId, object]):
+		registers[self.result.operandId] = registers[self.left.operandId] - registers[self.right.operandId]
+
+
+@dataclass
+class BinaryMultiply(_TypedInstruction):
+	opCode = lowLevel.InstructionType.BinaryMultiply
+	result: builder.Operand
+	left: builder.Operand
+	right: builder.Operand
+
+	def localExecute(self, registers: dict[lowLevel.OperandId, object]):
+		registers[self.result.operandId] = registers[self.left.operandId] * registers[self.right.operandId]
+
+
+@dataclass
+class BinaryDivide(_TypedInstruction):
+	opCode = lowLevel.InstructionType.BinaryDivide
+	result: builder.Operand
+	left: builder.Operand
+	right: builder.Operand
+
+	def localExecute(self, registers: dict[lowLevel.OperandId, object]):
+		left = registers[self.left.operandId]
+		right = registers[self.right.operandId]
+		if isinstance(left, int) and isinstance(right, int):
+			result = left // right
+		else:
+			result = left / right
+		registers[self.result.operandId] = result
+
+
+@dataclass
+class InplaceAdd(_TypedInstruction):
+	opCode = lowLevel.InstructionType.Add
+	target: builder.Operand
+	value: builder.Operand
+
+	def localExecute(self, registers: dict[lowLevel.OperandId, object]):
+		registers[self.target.operandId] += registers[self.value.operandId]
+
+
+@dataclass
+class InplaceSubtract(_TypedInstruction):
+	opCode = lowLevel.InstructionType.Subtract
+	target: builder.Operand
+	value: builder.Operand
+
+	def localExecute(self, registers: dict[lowLevel.OperandId, object]):
+		registers[self.target.operandId] -= registers[self.value.operandId]
+
+
+@dataclass
+class InplaceMultiply(_TypedInstruction):
+	opCode = lowLevel.InstructionType.Multiply
+	target: builder.Operand
+	value: builder.Operand
```

Comment:
or would it be

```suggestion
	target: builder.Operand[RemoteInt | RemoteFloat]
	value: builder.Operand[RemoteInt | RemoteFloat]
```

Reply:
I think it could actually be
```suggestion
	target: RemoteNumber
	value: RemoteNumber
```
Or yes, in theory we could use a TypeVar limited to RemoteInt, RemoteUint or RemoteFlaot, and use the Typevar for both, thus then the target and value would have to also be the same type. However, I think that Dataclass may actually use type annotations internally at runtime for docstring generation or validation, thus I don't think it would support using generics.  I could play with it, but at a minimum I could use RemoteNumber.  
I had avoided using higher-level remote types in these instructions as I wanted to keep them lower level and more true to how they are specified in the remote operations documentation. I.e. they are just operands. But I could try using remote types for all of them.

Reply:
More specific typing is nice if possible, but I'd consider this a nice to have

Reply:
Is it possible to add better typing instead of `builder.Operand` to instructions e.g. for target, value, etc?

Reply:
Unfortunately it causes circular imports to try and import and use remoteTypes such as `remoteTypes.RemoteString` on instructions such as `instructions.string.IsString` because RemoteString would be used on an annotation of the IsString instruction class at class creation time, and  the string instruction is referenced on RemoteString at class creation time. 
There may be a nice way to fix this - I agree it would be really nice if we could. But I fear I may need to refactor things quite a bit to make that work. Thus I would prefer not to block this pr on that as things are functioning well currently, and perhaps do this as future work.

Reply:
Have you tried/considered using `typing.TYPE_CHECKING` to only import the types when checking, or is that not possible in this case?

Reply:
Because these classes are DataClasses, I think that the type annotations are required at runtime.